### ISSUE TITLE
Error: incompatibility found between background and query lists, please set/upload a different background file or use a different gene list

### ISSUE THREAD

**Issue by Jin005** [dbitton/AnGeLi#10]
I pasted in my gene ID list and got this Error: incompatibility found between background and query lists, please set/upload a different background file or use a different gene list.

My gene ID list is from SGA data which fits in the non-essential genes, so I have chosen the 'pre-defined background' as so.

**Comment by dbitton** [comment]
Hi Jin, 

This error means that there is no overlap between the background list and the query list that you paste
If that is not the case, it is possible that the format of your background file is incorrect. Place each ID on a separate line, save as a tab delimited file (if you save file from Mac Excel, please use the 'Windows formatted Text .txt' option, please avoid any ctrl characters. If these do not solve the problem, please send me both lists to my UCL email and I can take a look.

**Comment by ValWood** [comment]
Jin, 

Also for the non-essential gene list you should use the total list in your experiment as the background (if the is the deletion collection it would probably be the non-essential sub-set from  4843 in PMID: 23697806).

For list validation, If your list is OK  you should be able to upload here:
 http://www.pombase.org/spombe/query/builder?filter=12

best wishes
Val

**Comment by dbitton** [comment]
Thanks Val and Jin,

We incorporate the non- essential genes as a background the last entry in the use pre-defined background

Cheers, 

Danny

**Comment by ValWood** [comment]
If I use all non essential genes as input (correct format) and select non-essential genes as background, I get an incompatability error. The lists must be mismatched.

This is the problem with predefining these 'essential' and 'non essential' lists. They are to dependent on many variables (time, source, dataset, conditions etc). Some deletions are annotated as both viable and inviable.

I would advise that  these options are removed, and people always  configure the background with the _exact list_ of genes that they used in their experiment.  If I was reviewing a paper with an enrichment I  would want to know _exactly_ which background set was used and the exact number of genes in it and that it matched the experiment.

**Comment by dbitton** [comment]
Hi Val, 

Not sure that I understand, the background of essential genes is being extracted directly from Pombase every update. Incompatibility error is reported when none of the genes in the query list are found in the background list, which in turn could be due to formatting issue or lack of intersection between background and query lists. Another unlikely option is that there is a very small intersection but these genes are missing from AnGeLi gene list.

I totally agree that background should be chosen properly and might be better than these defined set, I might found away to allow GO component to be search simultaneously, I will try to hack the code during this weekend

**Comment by ValWood** [comment]
Yes, but how do you extract the essential genes?  Do you take into account expression or conditions for example? Whichever list you use it is unlikely to exactly match the researchers background  (i.e experimental set). This means, that potentially the users results could always contain one or two genes which have been excluded from the selected background, hence the incompatibility error.

To test this I suggest that Jin find out exactly _which_ genes were used in the SGA array, and uses this as background. If this works it is likely that my assumption is correct.

**Comment by dbitton** [comment]
Yes I agree, I simply take all genes flagged as non essential, regardless of condition, I will highlight these issues in the help file

**Comment by ValWood** [comment]
which FYPO term do you use for this?

**Comment by ValWood** [comment]
You know, I really think it would be better not to have preconfigured background sets at all. It can really mislead people. I'm against any background configuration in principal, but all gene products and all gene proteins  as a default isn't too bad as long as it is really clear that these should  only be used in exceptional circumstances.

**Comment by dbitton** [comment]
I think I use this file ftp://ftp.ebi.ac.uk/pub/databases/pombase/pombe/Phenotype_annotations/FYPOviability.tsv and only viable and none condition-specific, but I have to check....

**Comment by dbitton** [comment]
Well I think that the PC and all genes are most used, but if one is only interested in FYPO or GO, I find these also useful, I agree that the non-essential is context dependent and better defined by users. I prefer not to remove any background at this stage, given that it is mentioned in the manuscript....

**Comment by ValWood** [comment]
OK, if you keep this option that is the best source. I forgot about this file, but it includes all viable population deletions (any condition). I would have expected any deletion which is ever viable to be in this list  so my original prediction may be wrong. It also depends if you  do with the ones which are flagged as "condition dependent" in column 2. If you discard these, that could be the problem.
